nullaway/src/main/java/com/uber/nullaway/dataflow/ArrayIndexElement.java

```diff
@@ -0,0 +1,42 @@
+package com.uber.nullaway.dataflow;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+import javax.lang.model.element.Element;
+
+public class ArrayIndexElement implements AccessPathElement {
+  private final Element javaElement;
+  @Nullable final String index;
```

Comment:
Why is this `@Nullable`?  Also, we shouldn't always use a `String` here.  We want to use something like an `Element` for a local variable I think.